dig_test.go

```diff
@@ -700,6 +700,59 @@ func TestEndToEndSuccess(t *testing.T) {
 			assert.Nil(t, p.Baz, "Baz must be unset")
 		}), "invoke failed")
 	})
+
+	t.Run("variadic arguments invoke", func(t *testing.T) {
+		c := New()
+
+		type A struct{}
+
+		var gaveA *A
+		require.NoError(t, c.Provide(func() *A {
+			gaveA = &A{}
+			return gaveA
+		}), "failed to provide A")
+
+		require.NoError(t, c.Provide(func() []*A {
+			panic("[]*A constructor must not be called.")
```

Comment:
I was sort of trying to stay consistent with existing tests that use panic.

Plus with `require.FailNow`, you now have to have an empty return statement to satisfy the compiler.

I can change this if you feel strongly about this.